src/hooks/useViewportRows.ts

```diff
@@ -1,6 +1,15 @@
 import { useMemo } from 'react';
 import { floor, max, min } from '../utils';
-import type { GroupRow, GroupByDictionary, RowHeightArgs } from '../types';
+import type { GroupRow, RowHeightArgs } from '../types';
+
+type GroupByDictionary<TRow> = Record<
+  string,
+  {
+    childRows: readonly TRow[];
+    childGroups: readonly TRow[] | Readonly<GroupByDictionary<TRow>>;
+    startRowIndex: number;
+  }
+>;
```

Comment:
Should work,, no?
```suggestion
type GroupByDictionary<TRow> = Readonly<Record<
  string,
  {
    readonly childRows: readonly TRow[];
    readonly childGroups: readonly TRow[] | GroupByDictionary<TRow>;
    readonly startRowIndex: number;
  }
>>;
```

Reply:
We [mutate](https://github.com/adazzle/react-data-grid/blob/472d2d27444b1251a59623005ac6b686c0fdb8c5/src/hooks/useViewportRows.ts#L59) the value but I can set fields to readonly